pkg/asset/installconfig/kubevirt/validation.go

```diff
@@ -0,0 +1,119 @@
+package kubevirt
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/kubevirt"
+	"github.com/openshift/installer/pkg/types/kubevirt/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// Validate executes kubevirt specific validation
+func Validate(ic *types.InstallConfig, clientBuilder ClientBuilderFuncType) error {
+	kubevirtPlatformPath := field.NewPath("platform", "kubevirt")
+
+	if ic.Platform.Kubevirt == nil {
+		return errors.New(field.Required(
+			kubevirtPlatformPath,
+			"validation requires a Engine platform configuration").Error())
+	}
+
+	return validatePlatform(ic.Platform.Kubevirt, ic.MachineNetwork, kubevirtPlatformPath, clientBuilder).ToAggregate()
+}
+
+func validatePlatform(kubevirtPlatform *kubevirt.Platform, machineNetworkEntryList []types.MachineNetworkEntry, fldPath *field.Path, clientBuilder ClientBuilderFuncType) field.ErrorList {
+	allErrs := validation.ValidatePlatform(kubevirtPlatform, fldPath)
+	ctx := context.Background()
+
+	client, resultErrs := validateInfraClusterReachable(ctx, kubevirtPlatform.Namespace, fldPath, clientBuilder)
+	allErrs = append(allErrs, resultErrs...)
+	if client != nil {
+		allErrs = append(allErrs, validateStorageClassExistsInInfraCluster(ctx, kubevirtPlatform.StorageClass, client, fldPath)...)
+		allErrs = append(allErrs, validateNetworkAttachmentDefinitionExistsInInfraCluster(ctx, kubevirtPlatform.NetworkName, kubevirtPlatform.Namespace, client, fldPath)...)
+	}
+	allErrs = append(allErrs, validateIPsInMachineNetworkEntryList(machineNetworkEntryList, kubevirtPlatform.APIVIP, kubevirtPlatform.IngressVIP, fldPath)...)
+
+	return allErrs
+}
+
+// validateInfraClusterReachable validates the following:
```

Comment:
The error handling inside won't tell for example that the namespace doesn't exists or kubevirt is intslled. Your intention is to return those error?

Reply:
It is required to support minimal permissions for the user
see the minimum required in this role: https://github.com/nirarg/installer/blob/master/docs/user/kubevirt/role.yml
Therefore, we can't perform getNamespace API call
Here I call to ListVM (in the namespace), which may fail in the following scenarios:
1.  Cluster is not reachable
2. KubeVirt is not installed and ListVM api doesn't exist
3. The namespace doesn't exist
4. User doesn't have permission to list vms

Reply:
Moved the comment that namespace is validated here, network-attachment-definition validation also validates the namespace exists